Updated README for first release (#281)
numtide/system-manager: README.md

</div>

System Manager brings the power of NixOS-style declarative configuration to other Linux distributions. Describe what your system should look like, by specifying packages, services, and settings all in Nix, then apply it safely and atomically with a single command. Each change is reproducible and soon will be rollback-ready, just like NixOS generations.

> If you're familiar with Home Manager, think of it as being similar, but for your entire machine. Whereas Home Manager manages user environments, System Manager manages the entire system, starting at root-level configurations, packages, and services, using the same reliable, Nix-based model.

Reviewer: Too long

System Manager builds on the many modules that already exist in [NixOS].

## Quick Example to Get Started

> We will assume you're using a non-NixOS distrubution (such as Ubuntu) and you have Nix already installed.

Reviewer: ```suggestion
We will assume you're using a non-NixOS distribution (such as Ubuntu) and you have Nix already installed, with flakes enabled.
```

Not a blocker but if we can assume that flakes are enabled, we can remove a bunch of edge-cases below.

Let's say we have a separate page that talks about installing Nix with flakes enabled, then we can keep linking back to it.

      };
    };
}
```

Then re-run System Manager and your changes will take effect; now you should have the two experimental features set, `nix-command` and `flakes`.

# Full Installation and setup

## Install Nix

System manager itself does not need to be installed; but, you do need to install Nix. (However, you can optionally install system-manager locally if you prefer.)

To install Nix, you can either use your distro's package manager, or use one of the following available options to install Nix.

- [Official Nix Installer][official-installer] - The canonical source for installing nix.
- [Determinate Nix Installer][detsys-installer] - A wrapper around the official installer that has
  SELinux support, and enables flake features by default.

> [!Tip]
> Some Nix users have had difficulty installing Nix on Windows Subsystem for Linux (WSL) using the official installer. If you're using WSL, we recommend using Determinate Nix installer.

## Example: Configuring System Services

The following example demonstrates how to specify a system service and activate it.

Update your flake.nix file to include a new file in the modules list, which we'll call `say_hello.nix`:

> ```

Reviewer: ````suggestion
```nix
````

</div>

> System Manager brings the power of NixOS-style declarative configuration to any Linux system. Describe what your system should look like, by specifying packages, services, and settings all in Nix, then apply it safely and atomically with a single command. Each change is reproducible and soon will be rollback-ready, just like NixOS generations.

Reviewer: ```suggestion
System Manager brings the power of NixOS-style declarative configuration to other Linux distributions. Describe what your system should look like, by specifying packages, services, and settings all in Nix, then apply it safely and atomically with a single command. Each change is reproducible and soon will be rollback-ready, just like NixOS generations.
```

> If you're familiar with Home Manager, think of it as similar to Home Manager but for your entire machine. Whereas Home Manager manages user environments, System Manager manages the full system, starting at root-level configurations, packages, and services, using the same reliable, Nix-based model.

Reviewer 2: ```suggestion
If you're familiar with Home Manager, think of it as being similar, but for your entire machine. Whereas Home Manager manages user environments, System Manager manages the entire system, starting at root-level configurations, packages, and services, using the same reliable, Nix-based model.
```

System Manager builds on the many modules that already exist in [NixOS].

## Quick Example to Get Started

Assume you're using a non-NixOS distrubution (such as Ubuntu) and you have Nix already installed.

In a folder create a file called `flake.nix` with the following:

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";

    system-manager = {
      url = "github:numtide/system-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = { self, nixpkgs, system-manager }: {
    systemConfigs.default = system-manager.lib.makeSystemConfig {
      modules = [

>         ./modules

Reviewer 3: Maybe we can make something more basic like:

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    system-manager = {
      url = "github:numtide/system-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs =
    {
      self,
      nixpkgs,
      system-manager,
    }:
    {
      systemConfigs.default = system-manager.lib.makeSystemConfig {
        modules = [
          (
            { config, pkgs, ... }:
            {
              nixpkgs.hostPlatform = "x86_64-linux";
              environment.systemPackages = with pkgs; [
                btop # Beautiful system monitor
                bat # Modern 'cat' with syntax highlighting
              ];
            }
          )
        ];
      };
    };
}
```

Author: Originally I did it this way, but somebody else in the project suggested I break it up so that the default.nix file bears some resemblance to what we use over in NixOS. I'm personally good with either, but not sure how to proceed... One person makes one suggestion, another person makes a different suggestion, and both seem equally good. 😄

      ];
    };
  };
}
```

In that folder create a subfolder called `modules`. In `modules` create a file called
`default.nix` with the following system configuration:

```nix
{ pkgs, ... }:
{
  nixpkgs.hostPlatform = "x86_64-linux";
  
  environment.systemPackages = with pkgs; [
    btop          # Beautiful system monitor
    bat           # Modern 'cat' with syntax highlighting
  ];
}

```

> This specifies a configuration that includes btop and bat as installed on the system. (In other words, system manager will install them on your system.) To do so, execute system manager with sudo using the nix command (assuming you have experimental features nix-command and flakes turned on):

Reviewer 2: ```suggestion
This specifies a configuration that includes `btop` and `bat`, to be installed on the system. To do so, execute system manager with `sudo` using the nix command (assuming you have experimental features nix-command and flakes turned on):
```

```
sudo env PATH="$PATH" nix run 'github:numtide/system-manager' -- switch --flake .
```

Notice we're passing the current PATH environment into sudo so that the elevated shell can locate the nix command.

Also, note that you might need to enable nix-commands and flakes:

```
sudo env PATH="$PATH" nix --extra-experimental-features 'nix-command flakes' run 'github:numtide/system-manager' -- switch --flake .
```

> [!Note]

> > The first time you run system manager, it will update your path by adding an entry inthe /etc/profile.d folder. For such change to take effect, you need to log out and log back in. However, if you don't want to log out, you can simply source the file:

Reviewer 2: ```suggestion
> The first time you run system manager, it will update your path by adding an entry in the /etc/profile.d folder. For this change to take effect, you need to log out and then log back in. However, if you don't want to log out, you can source the file:
```
